apps/styleguide/src/tables/RowEditModal.vue

```diff
@@ -152,87 +139,123 @@ export default {
-              (typeof this.value[column.id] === "number" &&
-                isNaN(this.value[column.id])))
-          ) {
-            this.errorPerColumn[column.id] = column.name + " is required ";
-          } else {
-            // when not empty
-            // when validation
-            if (
-              typeof this.value[column.id] !== "undefined" &&
-              typeof column.validation !== "undefined"
-            ) {
-              let value = this.value[column.id]; //used for eval, two lines below
-              this.errorPerColumn[column.id] = value; //dummy assign
-              this.errorPerColumn[column.id] = this.eval(column.validation);
-            } else if (
-              column.refLink &&
-              this.value[column.id] &&
-              this.value[column.refLink.replace(" ", "_")] &&
-              !JSON.stringify(this.value[column.id]).includes(
-                JSON.stringify(this.value[column.refLink.replace(" ", "_")])
-              )
-            ) {
-              //reflinks should overlap
-              this.errorPerColumn[column.id] =
-                "value should match your selection in column '" +
-                column.refLink +
-                "' ";
-            }
-          }
-        });
+        this.tableMetadata.columns.forEach((column) =>
+          this.validateColumn(column)
+        );
+      }
+    },
+    validateColumn(column) {
+      delete this.errorPerColumn[column.id];
+      const isInvalidNumber =
+        typeof this.value[column.id] === 'number' &&
+        isNaN(this.value[column.id]);
+      const isColumnValueInvalid = // how about undefined?
+        this.value[column.id] == null || isInvalidNumber;
+      if (column.required && isColumnValueInvalid) {
+        this.errorPerColumn[column.id] = column.name + ' is required ';
+      } else {
+        if (this.value[column.id] !== undefined && column.validation) {
+          this.evaluateValidationExpression(column);
+        } else if (this.isRefLinkWithoutOverlap(column)) {
+          this.errorPerColumn[column.id] =
+            "value should match your selection in column '" +
+            column.refLink +
+            "' ";
+        }
+      }
+    },
+    evaluateValidationExpression(column) {
+      try {
+        if (!Expressions.evaluate(column.validation, this.value)) {
+          this.errorPerColumn[
+            column.id
+          ] = `Error evaluating template: ${column.validation}`;
+        }
+      } catch (error) {
+        this.errorPerColumn[column.id] = `Invalid validation expression`;
       }
     },
+    isRefLinkWithoutOverlap(column) {
+      if (!column.refLink) {
+        return false;
+      } else {
+        const refLinkId = getRefLinkColumnByName(
+          this.tableMetadata,
+          column.refLink
+        ).id;
+        const value = this.value[column.id];
+        const refValue = this.value[refLinkId];
+
+        // TODO: find out if this should check for equality instead of overlap
```

Comment:
@mswertz Do you know why this does a .includes instead of using 'string'.equals('otherString')?